### PR TITLE
Use Charset instead of charset name in order to remove bottlenecks

### PR DESCRIPTION
**Pull request by jhanders34**
- When you pass in the charset name instead of the Charset itself, it ends up needing to look up the Charset object.  This can become a bottleneck in the JDK because the lookup ends up locking and if all threads are using the same charset we end up having threads waiting on each other.
- This change is similar to other changes done with PR #15997 and #16633

### CODE REVIEW

**Comment by jhanders34** [comment]
#build

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java [hunk 1]
```diff
@@ -338,7 +338,7 @@ protected void setPseudoHeaders(HashMap<String, String> pseudoHeaders) throws Ex
             this.setScheme(pseudoHeaders.get(HpackConstants.SCHEME));
         }
         if (pseudoHeaders.containsKey(HpackConstants.AUTHORITY)) {
-            parseH2Authority(pseudoHeaders.get(HpackConstants.AUTHORITY).getBytes());
+            parseH2Authority(pseudoHeaders.get(HpackConstants.AUTHORITY).getBytes(GenericUtils.CHARSET_FOR_GETBYTES));
```
Comment (jhanders34) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java]: Can this just be changed to:
```suggestion
            parseH2Authority(GenericUtils.getBytes(pseudoHeaders.get(HpackConstants.AUTHORITY));
```
and then just undo the changes in GenericUtils?

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/GenericUtils.java
Comment (loriadi) [inline dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/GenericUtils.java:1]: Copyright

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java (hunk 1 above)
Comment (loriadi) [reply dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java]: Unsure

**Review by loriadi — CHANGES_REQUESTED** [review]
I've been running with these changes all day with stress and have noticed nothing unusual.

File: dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/util/EncodingUtils.java [hunk 2]
```diff
@@ -479,31 +394,33 @@ public static String getJvmConverter(String encoding) {
...
-            if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable (Level.FINE)) {
-                logger.logp(Level.FINE, CLASS_NAME,"isCharsetSupported", "Encountered UnsupportedEncoding charset [" + charset +"]");
+    public static Charset getCharsetForName(String name) {
+        Charset charset = supportedEncodingsCache.get(name);
+        if (charset == null) {
+            try {
+                charset = Charset.forName(name);
+                supportedEncodingsCache.put(name, charset);
+            } catch (UnsupportedCharsetException e) {
+                if (com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled() && logger.isLoggable(Level.FINE)) {
```
Comment (pmd1nh) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/util/EncodingUtils.java]: Should this log be outside before the return? Otherwise, it will display once on the first set.

Comment (jhanders34) [reply dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/util/EncodingUtils.java]: I updated it to move the trace to be outside of the catch block.

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java (hunk 1 above)
Comment (jhanders34) [reply dev/com.ibm.ws.transport.http/src/com/ibm/ws/http/channel/internal/HttpRequestMessageImpl.java]: Based off of what I am reading like other header values, it is just characters 0 to 127 so I made the change as I put above.

File: dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/GenericUtils.java
Comment (jhanders34) [reply dev/com.ibm.ws.transport.http/src/com/ibm/ws/genericbnf/internal/GenericUtils.java:1]: Since the change isn't needed any longer, no update needed to the copyright since the file is removed from the PR now.

**Comment by jhanders34** [comment]
#libby #build

**Review by pmd1nh — APPROVED** [review]
WebContainer changes look good.

**Review by loriadi — APPROVED** [review]
good, thanks, Jared!
